src/util.py

```diff
@@ -1,10 +1,24 @@
 import yaml
+import argparse
 
 def load_hparams(path):
     with open(path, 'r') as stream:
         try:
-            defaults = yaml.safe_load(stream)['hparams']
+            hparams = yaml.safe_load(stream)['hparams']
         except yaml.YAMLError as e:
             print(e)
 
-    return defaults
+    return hparams
+
+
+def parse_train_args(args, hparams):
+    parser = argparse.ArgumentParser(description='Start training of the model')
+    parser.add_argument('--save_dir', metavar='PATH', help='Set the save directory for images and checkpoints', type=str, nargs='?', default=hparams['save_dir'])
```

Comment:
Again, `nargs='?'` is probably not what you want, since that means `python train.py --epochs` without a number would be accepted.